### DISCUSSION TITLE
Install help

### DISCUSSION THREAD

**Discussion by pturner890** [OpenRadioss/OpenRadioss#869] [Q&A]
Hello All,

Apologies if this is an easy one. Being quite new to linux, the installation can be quite difficult.

I am getting an error "reader configuration files are not available. Check and set RAD_CFG_PATH variable.    Standard configuration file installation is $ALTAIR_HOME/hwsolvers/radioss/cfg

I am trying to set all the RAD_CFG_PATH etc etc within a script file. I am certainly missing something obvious but I'm not sure what.

Many thanks for any help you can give!

Paul

**Comment by pturner890** [comment]
[script.txt](https://github.com/OpenRadioss/OpenRadioss/files/10124806/script.txt)

This is the script, feel like something is off with it but not sure what :(

**Reply by PaulAltair** [reply]
[ptscript.txt](https://github.com/OpenRadioss/OpenRadioss/files/10134288/ptscript.txt)
I set my variables in my .bashrc so they are always set in each terminal window, but I just tried your way and I think you need to end each line with ; and it should work, my version attached

**Reply by PaulAltair** [reply]
Also, Windows executables are available if you would find that easier!

**Comment by pturner890** [comment]
Paul,

Many thanks for your reply! I installed it on my laptop and had great
success - not sure if its just the version of ubuntu which was not very
happy (our analysis machine doesn't see the internet annoyingly and has a
slightly older version installed) or if I had slotted the
environment commands in the wrong part of bashrc...

Either way, very exciting!

Is there a user guide for the windows installation? Seems to be less click
and go than my primitive brain usually enjoys. In particular, if there is
one for offline installation.

Thanks again,

Paul

**Reply by PaulAltair** [reply]
The steps for windows are in [INSTALL.md](https://github.com/OpenRadioss/OpenRadioss/blob/main/INSTALL.md)
There is no 'install' to do as such, you just download and extract the package from a zip (can move to an offline machine on a USB I guess)
To run a model, the way I do it is open a cmd window local to my model and run a bat file for the envvars (similar to how you are doing it under linux) and then run another bat to run the job (just to save me remembering what to type!), my bats (renamed to .txt) are attached, you should just need to update the paths in them to wherever you have extracted the OpenRadioss package on your machine (this is just for SMP running, I haven't tried MPI on windows yet)
[envvars.txt](https://github.com/OpenRadioss/OpenRadioss/files/10142742/envvars.txt)
[runjob.txt](https://github.com/OpenRadioss/OpenRadioss/files/10142743/runjob.txt)

**Comment by pturner890** [comment]
Hi Paul,

Happy new year. Thank you very much for this - all up and running now.
Exciting times!

I was hoping I could pester you again slightly... the bumper model ran
perfectly. Great start, using a script to write out a bunch of output files
for paraview - beautiful.

I've tried a few of the vehicle models and they seem to start well, then it
crashes out - I suspect without any indication of why (I say suspect
because I felt I went through the person-readable files and nothing seemed
to indicate an issue).

I'm tempted to say it's a timestep issue, but I have just checked that and
it still doesn't work for timesteps much shorter than the default one. I
would hope that something close to the quoted one would be workable!

my yarisopenradioss.key file is a little modified, tried to add a timestep
control on there - that didnt work so I modified the combine.key and that
did the trick (as in the control worked). Sadly the model still crashed
out. I have attached a whole bunch of files some of which I hope are useful.

If you could provide any help I would be very grateful! This is the
Yariscoarse model but I tried a couple of the other options, coarse and
full and the other vehicle.

Many thanks,

Paul

**Reply by PaulAltair** [reply]
Hi Paul, I don't see any attachments? The Yaris should work completely unmodified using the YarisOpenRadioss.key starter file... but I haven't run it since the new year and sometimes an update will break something, I've resubmitted tonight (but it is queued) to check here, Sometimes models can be architecture sensitive (the Camry is) so that is another possibility, ping me tomorrow and we can take a look. Hopefully my run will have run (or not) by the morning as another data point.

**Reply by PaulAltair** [reply]
oh, and I found the actual problem with your runscript in the end, it was formatted with DOS line endings, switched to UNIX it worked fine

**Comment by pturner890** [comment]
ah! Thanks; such a rookie error. I'm very happy to abort using anything
linux based :)

did the files transfer through? I had a ping back from one of the email
addresses but I hope at least one of them worked.

Thanks again

Paul

**Reply by PaulAltair** [reply]
I haven't received anything, where are you sending them?

**Comment by pturner890** [comment]
[yaris.zip](https://github.com/OpenRadioss/OpenRadioss/files/10393009/yaris.zip)

Was replying to email, I'll drop them into the github discussion - they should be attached here now in a zip file

**Reply by PaulAltair** [reply]
ah, ok, I've not tried the 'coarse' Yaris, only the 'detailed' one, let me give that a go here too and I'll report back!

**Reply by PaulAltair** [reply]
I was able to run both detailed and coarse Yaris to completion (under Linux) using the YarisOpenRadioss.key launch file, the 1.out file you sent, just 'stops' no sign of a model issue or crash/seg fault in it or anything, not sure what issue may be there, I'll try submitting on windows too (but only have a laptop to try that on...)

**Reply by PaulAltair** [reply]
hmm, on Windows, job is exiting after a few hundred cycles (similar to what you have) will see if I can work out what issue is, it isn't a model issue specifically, same model is working fine in Linux. But I am able to run other models under windows so some combination of model and windows exe.

**Reply by pturner890** [reply]
Hi Paul - thanks again for looking into this. I tried the coarse and a detailed one, and the Camry model (all behaved in a similar way) Something to do with windows; not easy when no error messages being kicked out!

The bumper impact worked perfectly on a windows run though - if this helps narrow things down.

**Reply by PaulAltair** [reply]
Yes, we are able to replicate for the full vehicle models under windows pre-compiled binaries, we are still investigating cause, will report back when we get it resolved.

**Comment by pturner890** [comment]
Thanks Paul :) looking forward to it!

**Comment by pturner890** [comment]
Hi Paul,

Hope you are well :) I was wondering if you had any luck with the vehicle
modelling running in windows?

Best regards

Paul

**Reply by PaulAltair** [reply]
Prompted by this, I just rechecked, seems today it is working ok in the intel mpi, impi (SPMD) version, but still seems to be an issue with SMP, looking into it now

**Reply by PaulAltair** [reply]
> Hi Paul, Hope you are well :) I was wondering if you had any luck with the vehicle modelling running in windows? Best regards Paul

As of todays build it is working in both SMP and SPMD versions of the pre-compiled windows executable
